### DISCUSSION TITLE
Writing strings to a dataset

### DISCUSSION THREAD

**Discussion by JonesRusty** [metno/hdf5-rust#34] [Q&A]
I am new to Rust but have used HDF5 previously (in C++).

I am trying to figure out how to write a dataset that contains a single (fixed-length) string.

I can't find any examples and cannot see how to do this.

**Comment by magnusuMET** [comment]
This writes a fixed unicode string of length 20:
```rust
use hdf5_metno as hdf5;
use hdf5_metno::H5Type;
use std::str::FromStr;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let file = hdf5::File::create("file.h5")?;

    type T = hdf5::types::FixedUnicode::<20>;

    let data = T::from_str("This string")?;

    let ds = file.new_dataset::<T>().create("string")?;

    ds.write_scalar(&data)?;

    Ok(())
}
```

**Reply by JonesRusty** [reply]
That is so nice of you to get back to me so quickly! Thanks a bunch.
